### PR TITLE
Support Bitbucket

### PR DESCRIPTION
**Pull request by loganhz**
Proposed changes
======
Support Bitbucket in Pipeline

Types of changes
======
New feature

Linked Issues
======
https://github.com/rancher/rancher/issues/15700
 

Further comments
======

### CODE REVIEW

**Comment by loganhz** [comment]
It is ready for review. We need to merge the backend PR and frontend PR together

File: lib/pipeline/addon/components/settings-git/template.hbs [hunk 1]
```diff
@@ -19,6 +23,25 @@
       <h1 class="pull-left">{{t 'authPage.header'}}</h1>
     </section>
 
+    <section>
+      <div class="row nav nav-boxes checked-active">
+        <a href="#" alt={{selectedOauthType}} class="col span-2 nav-box-item driver active">
+          {{#if (eq selectedOauthType 'github')}}
```
Comment (westlywright) [inline lib/pipeline/addon/components/settings-git/template.hbs]: For both this and gitlab you check the param directly and use the `eq` helper. Wouldn't it make sense to split these into `isGithub` and `isGitlab`?

File: lib/pipeline/addon/components/settings-git/template.hbs [hunk 2]
```diff
@@ -43,161 +66,47 @@
...
-      <div class="row mt-30 text-center">
-        {{#if confirmDisable}}
-          {{#if disabling}}
-            <button class="btn bg-disabled" disabled="disabled">
-              <i class="icon icon-spinner icon-spin"></i> {{t 'generic.saving'}}
-            </button>
-          {{else}}
-          <button class="btn bg-primary" {{action "disable"}}>
-            <i class="icon icon-alert"></i> {{t 'pipelinesSetting.sureToDisableOAuth'}}
+      {{#if (eq provider.name 'bitbucketcloud')}}
```
Comment (westlywright) [inline lib/pipeline/addon/components/settings-git/template.hbs]: check `isBitbucket`?

File: lib/pipeline/addon/components/settings-git/template.hbs [hunk 3]
```diff
@@ -43,161 +66,47 @@
...
+
+      {{#if provider.clientId}}
+        <div>
+          <b>{{t (if isBitbucket 'authPage.authenticated.header.key.text' 'authPage.authenticated.header.clientId.text')}} </b>
+          <span class="text-muted">{{provider.clientId}}</span>
+        </div>
+      {{/if}}
+    </section>
+
+    <div class="row mt-30 text-center">
```
Comment (westlywright) [inline lib/pipeline/addon/components/settings-git/template.hbs]: Can you use {{save-cancel}} component here?

File: lib/pipeline/translations/en-us.yaml [hunk 4]
```diff
@@ -170,7 +173,7 @@ authPage:
       labelText: Enterprise Host
       placeholder: 'e.g. example.com'
   gitlab:
-    auth: 'For standard GitLab, <a href="https://gitlab.com/profile/applications" target="_blank">click here</a> to go applications settings in a new window.'
+    auth: 'For standard GitLab, <a href="https://gitlab.com/profile/applications" target="_blank">click here</a> to go to applications settings in a new window.'
```
Comment (westlywright) [inline lib/pipeline/translations/en-us.yaml]:
```suggestion
    auth: 'For standard GitLab, <a href="https://gitlab.com/profile/applications" target="_blank" rel="nofollow noreferrer">click here</a> to go to applications settings in a new window.'
```

File: lib/pipeline/translations/en-us.yaml [hunk 5]
```diff
@@ -187,7 +190,7 @@ authPage:
       clientSecret:
         labelText: Secret
   github:
-    auth: 'For standard GitHub, <a href="https://github.com/settings/developers" target="_blank">click here</a> to go applications settings in a new window.'
+    auth: 'For standard GitHub, <a href="https://github.com/settings/developers" target="_blank">click here</a> to go to applications settings in a new window.'
```
Comment (westlywright) [inline lib/pipeline/translations/en-us.yaml]:
```suggestion
    auth: 'For standard GitHub, <a href="https://github.com/settings/developers" target="_blank" rel="nofollow noreferrer">click here</a> to go to applications settings in a new window.'
```

File: lib/pipeline/translations/en-us.yaml [hunk 6]
```diff
@@ -205,7 +208,58 @@ authPage:
...
       clientSecret:
         labelText: Client Secret
-
+  bitbucketcloud:
+    target: 
+      header: '1. Choose the Bitbucket type you want to use'
+      bitbucketserver: 'Use private Bitbucket Server steup'
+      bitbucketcloud: 'Use public Bitbucket Cloud'
+    setup: '2. Setup a Bitbucket Cloud application'
+    auth: 'For Bitbucket Cloud, <a href="https://bitbucket.org/account" target="_blank">click here</a> to go to user settings in a new window.'
```
Comment (westlywright) [inline lib/pipeline/translations/en-us.yaml]:
```suggestion
    auth: 'For Bitbucket Cloud, <a href="https://bitbucket.org/account" target="_blank" rel="nofollow noreferrer">click here</a> to go to user settings in a new window.'
```

File: lib/pipeline/translations/zh-hans.yaml [hunk 7]
```diff
@@ -206,7 +207,54 @@ authPage:
...
       clientSecret:
         labelText: 客户端秘钥
-
+  bitbucketcloud:
+    target: 
+      header: '1. 选择你要使用的Bitbucket类型'
+      bitbucketserver: '使用私有Bitbucket Server'
+      bitbucketcloud: '使用Bitbucket Cloud'
+    setup: '2. 配置Bitbucket Cloud应用'
+    auth: '使用Bitbucket Cloud时，<a href="https://bitbucket.org/account" target="_blank">点击此处</a>，在弹出的新窗口中进行应用设置。'
```
Comment (westlywright) [inline lib/pipeline/translations/zh-hans.yaml]:
```suggestion
    auth: '使用Bitbucket Cloud时，<a href="https://bitbucket.org/account" target="_blank" rel="nofollow noreferrer">点击此处</a>，在弹出的新窗口中进行应用设置。'
```

File: lib/pipeline/addon/components/settings-git/template.hbs (hunk 3 above)
Comment (loganhz) [reply lib/pipeline/addon/components/settings-git/template.hbs]: There is a `Confirm Disable` step, so I guess I can not you `{{save-cancel}}` here

**Comment by loganhz** [comment]
@westlywright Thanks for the review!
Fixed and updated

**Pull request merged**